go/multiorch/recovery/actions/fix_replication.go

```diff
@@ -0,0 +1,356 @@
+	}
+
+	// Find a healthy primary in the shard
+	primary, err := a.poolerStore.FindHealthyPrimary(ctx, poolers)
+	if err != nil {
+		return mterrors.Wrap(err, "failed to find primary")
+	}
+
+	a.logger.InfoContext(ctx, "found primary for replication",
+		"primary", primary.MultiPooler.Id.Name,
+		"replica", replica.MultiPooler.Id.Name)
+
+	// Re-verify the problem still exists
+	needsFix, _, err := a.verifyProblemExists(ctx, replica, primary)
+	if err != nil {
+		return mterrors.Wrap(err, "failed to verify replication status")
+	}
+	if !needsFix {
+		a.logger.InfoContext(ctx, "replication already configured correctly, problem resolved",
+			"shard_key", problem.ShardKey.String(),
+			"pooler", problem.PoolerID.Name)
+		return nil
+	}
+
+	// Dispatch to the appropriate fix based on the problem
+	switch problem.Code {
+	case types.ProblemReplicaNotReplicating:
+		return a.fixNotReplicating(ctx, replica, primary)
+
+	// TODO: Future problem codes to handle
+	// case types.ProblemReplicaWrongPrimary:
+	//     return a.fixWrongPrimary(ctx, replica, primary, currentStatus)
+	// case types.ProblemReplicaLagging:
+	//     return a.fixReplicaLagging(ctx, replica, primary, currentStatus)
+	// case types.ProblemReplicaMisconfigured:
+	//     return a.fixMisconfigured(ctx, replica, primary, currentStatus)
+
+	default:
+		return mterrors.Errorf(mtrpcpb.Code_INVALID_ARGUMENT,
+			"unsupported problem code for fix replication: %s", problem.Code)
+	}
+}
+
+// fixNotReplicating handles the case where replication is not set up at all.
+// This is the most basic case: the replica has no primary_conninfo configured.
+func (a *FixReplicationAction) fixNotReplicating(
+	ctx context.Context,
+	replica *multiorchdatapb.PoolerHealthState,
+	primary *multiorchdatapb.PoolerHealthState,
+) error {
+	a.logger.InfoContext(ctx, "fixing replication: not configured",
+		"replica", replica.MultiPooler.Id.Name,
+		"primary", primary.MultiPooler.Id.Name)
+
+	// Get the primary's postgres port
+	primaryPort, ok := primary.MultiPooler.PortMap["postgres"]
+	if !ok {
+		return mterrors.Errorf(mtrpcpb.Code_FAILED_PRECONDITION,
+			"primary %s has no postgres port configured", primary.MultiPooler.Id.Name)
+	}
+
+	// Get the current consensus term from the primary
+	consensusResp, err := a.rpcClient.ConsensusStatus(ctx, primary.MultiPooler, &consensusdatapb.StatusRequest{})
+	if err != nil {
+		return mterrors.Wrap(err, "failed to get consensus status from primary")
+	}
+
+	consensusTerm := consensusResp.CurrentTerm
+
+	a.logger.InfoContext(ctx, "got consensus term from primary",
+		"primary", primary.MultiPooler.Id.Name,
+		"consensus_term", consensusTerm)
+
+	// Configure primary_conninfo on the replica
+	req := &multipoolermanagerdatapb.SetPrimaryConnInfoRequest{
+		Host:                  primary.MultiPooler.Hostname,
+		Port:                  primaryPort,
+		StopReplicationBefore: true,
+		StartReplicationAfter: true,
+		CurrentTerm:           consensusTerm,
```

Comment:
This seemed sensible, but we should discuss. The way `SetPrimaryConnInfoRequest` works today, requires you to pass a valid term, otherwise it will reject the request.